### DISCUSSION TITLE
what files make up the runtime?

### DISCUSSION THREAD

**Discussion by usernamedd** [dotnet/runtime#67507] [General] [answered]
It is not about a but . I want to dig deep to runtime files .  or the question may be like : when we install a dotnet runtime , what exactly the windows does , how many files are released and what about the registry?

**Comment by ghost** [comment]
I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one [area label](https://github.com/dotnet/runtime/blob/master/docs/area-owners.md).

**Comment by ghost** [comment]
Tagging subscribers to this area: @dotnet/area-meta
See info in [area-owners.md](https://github.com/dotnet/runtime/blob/main/docs/area-owners.md) if you want to be subscribed.
<details>
<summary>Issue Details</summary>
<hr />

It is not about a but . I want to dig deep to runtime files .  or the question may be like : when we install a dotnet runtime , what exactly the windows does , how many files are released and what about the registry?

<table>
  <tr>
    <th align="left">Author:</th>
    <td>usernamedd</td>
  </tr>
  <tr>
    <th align="left">Assignees:</th>
    <td>-</td>
  </tr>
  <tr>
    <th align="left">Labels:</th>
    <td>

`area-Meta`, `untriaged`

</td>
  </tr>
  <tr>
    <th align="left">Milestone:</th>
    <td>-</td>
  </tr>
</table>
</details>

**Comment by danmoseley** [comment]
@usernamedd could you say more about what you're trying to achieve?

When you install .NET Core (eg .NET 6) it installs into %programfiles%\dotnet, on Linux it installs to /usr/share/dotnet etc. As far as I know, on Windows it does not write to the registry except what is necessary for install/uninstall.

**Reply by usernamedd** [reply]
Thank you.
out of curiosity . If no infomation is writen to registry , how does windows loader know where to find the files our exe depends . As far as I know , when the loader finds the exe is managed , then the loader will load runtime ,i.e the coreclr.dll . Question is how the loader gets the coreclr.dll.

**Reply by vitek-karas** [reply]
Related designs for that:
https://github.com/dotnet/designs/blob/main/accepted/2020/install-locations.md
https://github.com/dotnet/designs/blob/main/accepted/2021/install-location-per-architecture.md
https://github.com/dotnet/designs/blob/main/accepted/2019/runtime-binding.md

**Comment by gfoidl** [comment, marked as answer]
> I want to dig deep to runtime files

As potential starting points:
* [The Book of the Runtime](https://github.com/dotnet/runtime/tree/main/docs/design/coreclr/botr#the-book-of-the-runtime)
* [List of blog posts that provide a 'deep-dive' into the CoreCLR source code](https://github.com/dotnet/runtime/blob/main/docs/deep-dive-blog-posts.md)

Enjoy it 😃

**Reply by usernamedd** [reply]
Thank a ton. that is really what I am looking for.

**Reply by danmoseley** [reply]
Since you mention loader, you may also be interested in the various documents about the host, in https://github.com/dotnet/runtime/tree/main/docs/design/features (host-*)

Windows has special knowledge of .NET Framework exe's. This is not the case for .NET Core exe's and they must either be run with dotnet.exe or specially prepared as a single file app.

**Reply by usernamedd** [reply]
Thx
